server/jobs/NetXSyncJob.js

```diff
@@ -0,0 +1,100 @@
+const ElasticSearchService = require("../services/elasticSearchService");
+const ObjectAssetService = require("../services/objectAssetService");
+const { DAMSService } = require("../services/damsService");
+
+const LOG_PREFIX = `[NetXSyncJob]`;
+
+function generateQueryInput({ start, size }) {
+  return {
+    body: {
+      sort: [
+        {
+          _id: {
+            order: "asc",
+          },
+        },
+      ],
+      from: start,
+      size: size,
+      _source: ["_id", "id", "invno", "imageOriginalSecret", "imageSecret"],
+      query: {
+        bool: {
+          filter: {
+            exists: {
+              field: "imageSecret",
+            },
+          },
+        },
+      },
+    },
+  };
+}
+
+/** Retrieves the Object IDs and Invno Numbers for all objects available in our ElasticSearch database
+ * We're retrieving these from ElasticSearch because ElasticSearch is the source against which our Collection Site
+ * performs queries against.
+ *
+ * While we do source and populate our ElasticSearch database using data from NetX, it is ultimately the Artwork records
+ * that exist in ElasticSearch that are the dataset we filter and query against to shown on the Collection Site.
+ *
+ * Therefore, it makes sense to retrieve the Object IDs list from there, and then use it to query against NetX
+ */
+async function main() {
+  const totalRecordCount = await ElasticSearchService.getCount();
+  console.log(`
+    ${LOG_PREFIX} There are a total of ${totalRecordCount} records in ElasticSearch
+    `);
+
+  const recordBatchSize = 250;
+  const numberOfBatches = Math.ceil(totalRecordCount / recordBatchSize);
+  console.log(
+    `Job will process ${recordBatchSize} in ${numberOfBatches} batches `
+  );
+
+  for (let i = 0; i <= numberOfBatches; i++) {
```

Comment:
Main body of the NetX Sync Job. Iterates through our ElasticSearch artwork records and retrieves them from NetX while caching them.

It then looks up all archive assets and augments the cache artwork entry with these additional archive assets.